src/uct/api/uct.h

```diff
@@ -265,6 +265,25 @@ enum uct_iface_event_types {
 
 /**
  * @ingroup UCT_RESOURCE
+ * @brief  Flush modifiers.
+ */
+enum uct_flush_flags {
+    UCT_FLUSH_FLAG_LOCAL    = 0,            /**< Guarantees that the data
+                                                 transfer is completed but the
+                                                 target buffer may not be
+                                                 updated yet.*/
+    UCT_FLUSH_FLAG_CANCEL   = UCS_BIT(1)    /**< Cancel all outstanding and
+                                                 pending operations.
```

Comment:
Potentially you may fail to cancel some of the operations , right ?  If so you have to be clear about this in the documentation

Reply:
Not completely understand your point. 
1. If operation is scheduled (in pending queue), it's no problem to cancel it.
2. If operation was posted but was not really completed, there are options:
    2.1 short or bcopy do not have completion callback and user assumes that operation is completed successfully by return code. It's his responsibility how to handle such behavior. if he wants to deliver all operation to receiver he should not use this flag.
    2.2 Zcopy has completion callback and user should get it with "cancel" status.

Reply:
You should have make assumptions about underlaying implementation of the transport 
(bcopy and zcopy can be implemented in hardware). You should state something more abstract. The library will do the best effort to cancel all in completed operations. User should be aware about the fact there there is a chance that some operation will not be completed and been ready to handle all the relevant callbacks.